core/org.wso2.carbon.core/src/main/java/org/wso2/carbon/core/multitenancy/MultitenantMessageReceiver.java

```diff
@@ -71,15 +71,19 @@
 	private static final Log log = LogFactory.getLog(MultitenantMessageReceiver.class);                      
 
     public void receive(MessageContext mainInMsgContext) throws AxisFault {
-
-        EndpointReference toEpr = getDestinationEPR(mainInMsgContext);
-        if (toEpr != null) {
-            // this is a request coming in to the multitenant environment
-            processRequest(mainInMsgContext);
-        } else {
-            // this is a response coming from a dual channel invocation or a
-            // non blocking transport like esb
-            processResponse(mainInMsgContext);
+        PrivilegedCarbonContext.getThreadLocalCarbonContext().startTenantFlow();
```

Comment:
Can you please explain the reason to start a tenant flow here? Even though a new tenant flow is started, the tenant information has not been set.